### DISCUSSION TITLE
How to limit the scope to search only the current line bi-directionally?

### DISCUSSION THREAD

**Discussion by kohane27** [ggandor/leap.nvim#185] [Q&A] [answered]
Hello there. Hope you're doing well. Thank you for creating `leap.nvim`; it completely changed my navigation style in Neovim.

I'm wondering how to extend the same functionality but for only the current line. Basically, `leap.nvim` but only for the current line.

1. Cursor at `xxx`

```
This is a test hello hello   xxx   hello hello
                              ^
                            cursor
```

2. Press `f`
3. Highlight all possible targets (in the above case, there are 4)
4. Press the label to jump to the target

Desired result:

![2023-09-12-15-18-17-leap](https://github.com/ggandor/leap.nvim/assets/57322459/a0c67b9a-a814-4c90-8612-026173a3ec7b)

## What I have Tried

```lua
local function leap_current_line()
  local current_line = vim.fn.line(".")
  print(vim.inspect(current_line))
  local targets = {}
  -- TODO: how to search all targets on current line?

  require("leap").leap({
    target_windows = { vim.fn.win_getid() },
    targets = targets,
  })
end
```

But I have trouble extending and finishing up the above function.

Any input is much appreciated. Thank you!

**Comment by ggandor** [comment, marked as answer]
It's doable if you don't want 2-step highlighting, but far from trivial. Otherwise you'd have to reimplement half the plugin, so it's practically impossible.

For the former, you might want to check flit.nvim:
https://github.com/ggandor/flit.nvim/blob/498b3a4864e697f1ed1145e518c4c78c776c55d8/lua/flit.lua#L10
https://github.com/ggandor/flit.nvim/blob/498b3a4864e697f1ed1145e518c4c78c776c55d8/lua/flit.lua#L38
https://github.com/ggandor/flit.nvim/blob/498b3a4864e697f1ed1145e518c4c78c776c55d8/lua/flit.lua#L55
